app/backend/wells/views.py

```diff
@@ -185,7 +185,7 @@ class WellTagSearchAPIView(ListAPIView):
 
     permission_classes = (DjangoModelPermissionsOrAnonReadOnly,)
     model = Well
-    queryset = Well.objects.only('well_tag_number', 'owner_full_name').all()
+    queryset = Well.objects.exclude(well_publication_status='Unpublished').only('well_tag_number', 'owner_full_name')
     pagination_class = None
```

Comment:
If a logged in user wants to edit a well, to make it published, the user would have to be able to find it. I think we should only apply this filter to users who aren't logged in with well edit rights.